go/private/actions/link.bzl

```diff
@@ -159,6 +159,8 @@ def _bootstrap_link(go, archive, executable, gc_linkopts):
         inputs = inputs,
         outputs = [executable],
         mnemonic = "GoLink",
+         # workaround: go link tool needs some features of gcc to complete the job on Arm platform.
```

Comment:
`PATH` isn't actually set here.

Instead of setting `PATH` though, I think it would be better to set `CC` to `go.cgo_tools.ld_executable`.

Reply:
Hi @jayconrod 
'CC=go.cgo_tools.ld_executable' is useless here.
I modified the code like this:
command = "export GOROOT=$(pwd)/{} && export GOROOT_FINAL=GOROOT && export CC={} && {} {}".format(go.root, go.cgo_tools.ld_executable, go.go.path, " ".join(args)),
And google/gvisor still reported that 'gcc' was not found  in $PATH.
Please see full logs as reference:
=============================
root@bin:/go/src/github.com/google/gvisor# bazel build runsc --verbose_failures
INFO: SHA256 (https://github.com/lubinsz/rules_go/archive/801a52883ef36ae078f069bc250a325ca7121acd.tar.gz) = 5ba38ec2b0bee96a77793f7f95669fa4ea101599aa39eb8dce1b9daf149431fd
INFO: Build options have changed, discarding analysis cache.
INFO: Analysed target //runsc:runsc (183 packages loaded).
INFO: Found 1 target...
ERROR: /root/.cache/bazel/_bazel_root/954c2359eaedb37da1533cf5fa671511/external/io_bazel_rules_go/go/tools/builders/BUILD.bazel:54:1: error executing shell command: 'export GOROOT=$(pwd)/external/go_sdk && export GOROOT_FINAL=GOROOT && export CC=/usr/bin/ld && external/go_sdk/bin/go tool link -s -o bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders...' failed (Exit 1): bash failed: error executing command
  (cd /root/.cache/bazel/_bazel_root/954c2359eaedb37da1533cf5fa671511/execroot/__main__ && \
  exec env - \
  /bin/bash -c 'export GOROOT=$(pwd)/external/go_sdk && export GOROOT_FINAL=GOROOT && export CC=/usr/bin/ld && external/go_sdk/bin/go tool link -s -o bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/linux_arm64_stripped/embed bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/linux_arm64_stripped/embed~/go/tools/builders/embed.a')

Use --sandbox_debug to see verbose messages from the sandbox
/root/.cache/bazel/_bazel_root/954c2359eaedb37da1533cf5fa671511/sandbox/linux-sandbox/4/execroot/__main__/external/go_sdk/pkg/tool/linux_arm64/link: running gcc failed: exec: "gcc": executable file not found in $PATH

Target //runsc:runsc failed to build
ERROR: /go/src/github.com/google/gvisor/runsc/BUILD:5:1 error executing shell command: 'export GOROOT=$(pwd)/external/go_sdk && export GOROOT_FINAL=GOROOT && export CC=/usr/bin/ld && external/go_sdk/bin/go tool link -s -o bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders...' failed (Exit 1): bash failed: error executing command
  (cd /root/.cache/bazel/_bazel_root/954c2359eaedb37da1533cf5fa671511/execroot/__main__ && \
  exec env - \
  /bin/bash -c 'export GOROOT=$(pwd)/external/go_sdk && export GOROOT_FINAL=GOROOT && export CC=/usr/bin/ld && external/go_sdk/bin/go tool link -s -o bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/linux_arm64_stripped/embed bazel-out/host/bin/external/io_bazel_rules_go/go/tools/builders/linux_arm64_stripped/embed~/go/tools/builders/embed.a')

Use --sandbox_debug to see verbose messages from the sandbox
INFO: Elapsed time: 20.737s, Critical Path: 8.50s
INFO: 7 processes, linux-sandbox.
FAILED: Build did NOT complete successfully
root@bin:/go/src/github.com/google/gvisor# uname -p
aarch64